### PR TITLE
Add support for escape() vim function

### PR DESCRIPTION
**Pull request by jphalip**

### CODE REVIEW

File: src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/functions/BuiltInFunctionTest.kt [hunk 1]
```diff
@@ -177,4 +177,28 @@ class BuiltInFunctionTest : VimTestCase() {
...
+    assertExOutput("no escaping needed")
+
+    // Test empty strings
+    typeText(commandToKeys("""echo escape('', '')"""))
+    assertExOutput("")
+
+    // Test consecutive characters that need escaping
+    typeText(commandToKeys("""echo escape('$$$$', '$')"""))
+    assertExOutput("""\$\$\$\$""")
+  }
```
Comment (citizenmatt) [inline src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/functions/BuiltInFunctionTest.kt:203]: It would be good to extract this to a separate test class, perhaps under a `builtin` folder, so we can have separate files for each function, and allow splitting the test into more logical tests/asserts.

Comment (citizenmatt) [reply src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/functions/BuiltInFunctionTest.kt:203]: Probably worth adding a couple of tests for invalid method calls - too many parameters, not enough parameters, what happens with invalid data types, etc.

**Comment by AlexPl292** [comment]
Thank you for your PR! Can you please take a look and write a test on how this would work with multicharacter symbols like `😀`?

**Review by AlexPl292 — CHANGES_REQUESTED** [review]
Please check the comments

**Comment by jphalip** [comment]
I've improved the implementation to support unicodes, emojis, etc. and added multiple other tests.

However, maybe this is overachieving? For example, standard vim appears to ignore escaping of emojis:

Standard vim:
```vim
:echo escape("😀🎉✨", "😀✨")
😀🎉✨
```

This implementation:
```kotlin
typeText(commandToKeys("""echo escape("😀🎉✨", "😀✨")"""))
assertExOutput("\\😀🎉\\✨")
```

Let me know if you're ok with this or if you'd like to scale back to the standard behavior.

File: src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/functions/BuiltInFunctionTest.kt (hunk 1 above)
Comment (jphalip) [reply src/test/java/org/jetbrains/plugins/ideavim/ex/implementation/functions/BuiltInFunctionTest.kt:209]: @citizenmatt I've added a couple of tests with errors.

I also agree that it'd be nice to separate these buitlin functions out to separate class, but maybe this should be done in bulk in a separate PR?

**Comment by AlexPl292** [comment]
Oh, sorry, I didn't mean THAT accurate :D
I mentioned `😀` because I thought that iterating over code points (via `String.codePoints()`) would generally solve the issue. Will working just with codepoints make the escaping more accurate? If not, let's use the original behavior because it looks much simpler and more stable :)

**Comment by AlexPl292** [comment]
Note: I closed & reopened the PR by accident

**Comment by jphalip** [comment]
@AlexPl292 I've reverted to my original, simpler implementation. I also added some tests for unicode, and verified that the results are the same in standard Vim.

**Comment by AlexPl292** [comment]
Great, thank you!
